test/test_helper.rb

```diff
@@ -16,8 +16,11 @@
 # <http://www.gnu.org/licenses/agpl.html>
 #
 require 'minitest'
-require 'simplecov'
-SimpleCov.start if (!ENV.has_key?('COV') && !ENV.has_key?('COVERAGE')) || (ENV['COV'] != 'false' && ENV['COVERAGE'] != 'false')
+
+if Rake.application.top_level_tasks.include?('test') && (!ENV.key?('COV') && !ENV.key?('COVERAGE')) || (ENV['COV'] != 'false' && ENV['COVERAGE'] != 'false')
```

Comment:
I disagree with this as when putting coverage to false, no-one will launch it with true.

Reply:
If all tests set it to `ENV['COV'] = 'false'` in the Rakefile then what's the point ;) no one could launch it with `true` in that case!

Plus the default behavior of this part of the code should be in accord with the default behavior of the other parts. The procedures we define should let the CI and the developers know when to launch the tests with coverage or not. 

For example, with my suggestions, the CI can launch the tests always with `COV=true` and we can have check that fails whenever there is a regression in the coverage. Forcing the coverage in this manner would only mean that the devs will launch the tests with `COV=false rake test` on local.

Reply:
All test are set to COV=true, only the specifics one set the coverage to false. The coverage is actually set to false in the CI so if it's the only place coverage is calculated, there is indeed some changes to do.

Reply:
As far as I know, the local tests are launched with an alias which has `COVERAGE=false`, that is, without the modification above the coverage is never launched. With the modification above, the `COV=true` which we define in the main test will be enforced correctly because we check "if there exists true" in the environment vars --  `[ENV['COVERAGE'], ENV['COV']].include? 'true'`.

I still think this sort of task is something that we can handle very efficiently with CI and it shouldn't be left as a manual verification on local. What do you think @alain-andre?

Reply:
for the moment, I prefer to keep coverage by default when replaying all tests (same thing for all projects), we will see other strategies when we will have an advanced CI